src/core/pages/Charts/index.js

```diff
@@ -15,6 +15,7 @@ import {
 import { withStyles } from '@material-ui/core/styles';
 
 import { patientStatisticAction } from "../../actions/patientsStatisticAction";
+import dummyPatients from "../PatientsList/dummyPatients";
```

Comment:
OK, but how in this case I can get patients data in chart file?

I did this:
https://github.com/PulseTile/PulseTile-RA/blob/feature-patients-list/src/core/pages/Charts/index.js

But when I try to ask data from dataProvider I don't pass to **return**:
```
// Charts/index.js
class Charts extends Component {
    componentDidMount() {
        customDataProvider(GET_LIST, 'patients', {});
    }

    render() {
          ....
    }
}

// dataProviders/dataProvider.js, I can't pass in this return:
    return (type, resource, params) => {
        if (resource === `patients`) {
            return fakePatientsProvider(type, resource, params);
        }
        return dataProvider(type, resource, params);
    };
```

Reply:
I don't understand what you mean by "I don't pass to return".

See the documentation about querying your API with react-admin (https://marmelab.com/react-admin/Actions.html):

Reply:
> I don't pass to return
I mean than I tried to debug my code:
1) Call dataProvider in Charts component (row 33):
https://github.com/PulseTile/PulseTile-RA/blob/feature-patients-list/src/core/pages/Charts/index.js
2) Wrote console.log('++++++') after the row 315 (inside return):
https://github.com/PulseTile/PulseTile-RA/blob/feature-patients-list/src/core/dataProviders/dataProvider.js

So, I passed to dataProvider, but don't pass to inside final return (....);

> See the documentation about querying your API with react-admin (https://marmelab.com/react-admin/Actions.html):

I tried to use decorator withDataProvider, but get a message:
_'react-admin' does not contain an export named 'withDataProvider'._

I use React-admin version 2.6.0 (the latest is 2.8.0). 
Is my version old?

Reply:
Sorry, I still don't understand. "I passed to dataProvider" is missing an object. You passed *what* to the dataProvider? "I don't pass to inside final return", what doesn't pass inside which function? what is your error? 

I see that `dataProvider/dataProvider.js` exports a function that returns a function (not sure why? It's a useless indirection). In order to actually call the dataProvider in the Charts component, you'd have to call 

`customDataProvider()(GET_LIST, 'patients', {})` 

instead of 

`customDataProvider(GET_LIST, 'patients', {})`

But you'd better remove the enclosing function in the data Provider.

withDataProvider is indeed a new feature, and requires react-admin 2.8.

Reply:
Done. I removed enclosing function in dataProvider.